merge_forward_pull_request.bash: new script

scpeters opened the pull request:
Resolves #671 by adding a script to simplify creation of forward port pull requests that uses the `gh` tool and automatically fills in the template.

I used this script to create https://github.com/ignitionrobotics/ign-math/pull/421 with the following procedure:

~~~
cd ign-math
git checkout main
git pull
git checkout -b merge67
git merge ign-math6
# manually resolve conflicts
git merge --continue
# then run this script
bash ../release-tools/release-repo-scripts/merge_forward_pull_request.bash ign-math6 main
~~~

scpeters commented:
it needs the message about using a merge commit at the end

chapulina commented on release-repo-scripts/merge_forward_pull_request.bash:
```diff
@@ -0,0 +1,53 @@
+ORIGIN_URL=$(git remote get-url origin)
+ORIGIN_ORG_REPO=$(echo ${ORIGIN_URL} | sed -e 's@.*github\.com.@@')
+
+TITLE="Merge ${FROM_BRANCH} -> ${TO_BRANCH}"
```
Can we use ➡️ here? I find it easier to spot when looking through the PR list.

```suggestion
TITLE="Merge ${FROM_BRANCH} ➡️ ${TO_BRANCH}"
```

scpeters replied:
https://github.com/gazebo-tooling/release-tools/pull/720/commits/97845bac0fe1b27d2fd26a5e171cd92c912f41a9

chapulina replied:
That's not in the title though 😢

scpeters replied:
sorry, it should be in the title as of https://github.com/gazebo-tooling/release-tools/pull/720/commits/b517bbdd8db92541f36a69c6cfc73de74934f254

chapulina commented on release-repo-scripts/merge_forward_pull_request.bash:
```diff
@@ -0,0 +1,53 @@
+# limitations under the License.
+#
+
+# The script will merge forward and open a pull request.
```
When I first read this I understood that the `git` calls would be handled by the script

```suggestion
# The script will open a pull request for a forward port.
```

It would be nice if these lines were handled by the script


```
git checkout main
git pull
git checkout -b merge67
git merge ign-math6
# manually resolve conflicts
git merge --continue
```

scpeters replied:
I applied your suggestion in https://github.com/gazebo-tooling/release-tools/pull/720/commits/d66e26c205908226e5361a3995cf6ea4595345ca

> It would be nice if these lines were handled by the script

I hear you but I'm not sure how to handle manually resolving conflicts with a script like this. I think it would be easier to have a separate script for the first few lines and then invoke this script afterwards once the conflicts are resolved. At any rate, I think that would be easier to handle in a follow-up pull request

chapulina replied:
A follow-up PR sounds good. 

One idea would be to stop and prompt the user to manually inspect the files and conflicts, similar to

https://github.com/gazebo-tooling/release-tools/blob/b213ef89f81bd4320e13c4ec06342613a28a26a4/release-repo-scripts/bump_dependency.bash#L194-L200

And when the user does `y`, then proceed with the `git merge --continue`.

chapulina reviewed (approved):
Works for me, used it to open 

* https://github.com/gazebosim/gz-tools/pull/88

As mentioned in https://github.com/gazebo-tooling/release-tools/pull/716#issuecomment-1124249347, I'd move this to `source-repo-scripts` and add some lines to `source-repo-scripts/RELEASE.md`, similar to  what's being done in #677

nuclearsandwich commented on release-repo-scripts/merge_forward_pull_request.bash:
```diff
@@ -0,0 +1,55 @@
+
+**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)"
+
+if [[ $# -lt 2 ]]; then
```
```suggestion
if [[ $# -ne 2 ]]; then
```

Shouldn't there be exactly two arguments rather than two or more?

chapulina commented:
Also used to open

* https://github.com/gazebosim/gz-sim/pull/1481

j-rivero commented on source-repo-scripts/README.md:
````diff
@@ -39,6 +39,35 @@ its dependencies on fortress using the `chapulina/fortress` branch for `docs`:
+To merge `ign-rendering6` to `main`:
+
+```bash
+cd ign-rendering
````
Maybe for future work but these steps could probably be part of a script too until the point of the manual intervention for the conflict resolution.

scpeters replied:
@chapulina suggested the same https://github.com/gazebo-tooling/release-tools/pull/720#discussion_r868323853, and we decided to address in a follow-up pull request

scpeters commented on source-repo-scripts/merge_forward_pull_request.bash:
```diff
@@ -0,0 +1,55 @@
+
+**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)"
+
+if [[ $# -ne 2 ]]; then
```
https://github.com/gazebo-tooling/release-tools/pull/720/commits/1093550f83bf107806662d2acea1fc0d0da8937e

scpeters commented:
I just used this again for the following:

* https://github.com/gazebosim/gz-sim/pull/1491

this has been approved and I believe I've addressed all comments, so I will merge
